precompiles/gmp/Gmp.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.3;
+
+/// @dev The Gmp contract's address.
+address constant GMP_ADDRESS = 0x0000000000000000000000000000000000000815;
+
+/// @dev The Gmp contract's instance.
+Gmp constant GMP_CONTRACT = Gmp(GMP_ADDRESS);
+
+/// @author The Moonbeam Team
+/// @title Gmp precompile
+/// @dev Provides an endpoint to Gmp protocols which can automatically forward to XCM
+/// @custom:address 0x0000000000000000000000000000000000000815
+interface Gmp {
+    // TODO: Here we would specify the endpoints for each GMP protocol on a case by case basis.
+    //       These endpoints are basically the hand offs for each protocol -- where they delegate to
+    //       the target contract.
+    //
+    //       This design should allow users to interact with this precompile with no changes to the
+    //       underlying GMP protocols by simply specifying the correct precompile as the target.
+
+    /// Receive a wormhole VAA and process it
```

Comment:
Even if the VAA is generic, we expect it to have some shape related to xtokens. Maybe we can specify it here?

Reply:
I'm not sure what you mean. Are you referring to the way a user specifies what XCM interaction (e.g. target MultiLocation for xtokens)? That would be defined in `types.rs` with `VersionedUserAction`, which does need some better documentation at some point...